plugins/user-authenticators/ldap/src/org/apache/cloudstack/ldap/LdapManagerImpl.java

```diff
@@ -243,4 +263,22 @@ public boolean isLdapEnabled() {
             closeContext(context);
         }
     }
+
+    @Override
+    public LinkDomainToLdapResponse linkDomainToLdap(Long domainId, String type, String name, short accountType) {
+        Validate.notNull(type, "type cannot be null. It should either be GROUP or OU");
+        Validate.notNull(domainId, "domainId cannot be null.");
```

Comment:
What if domainId is already present? I understand there will be a DB constraint violation exception but it is better to throw InvalidParameterException instead.